src/Shells/StrixMusic.Shells.ZuneDesktop/Controls/Views/Collection/ZuneTrackCollection.cs

```diff
@@ -144,6 +199,14 @@ private async Task OnCollectionChangedAsync(ITrackCollectionViewModel? oldValue,
             {
                 oldValue.Tracks.CollectionChanged -= Tracks_CollectionChanged;
             }
+
+            // Giving breathing space to cpu.
+            await Task.Delay(10);
```

Comment:
If you have to use Task.Delay, you're most likely patching over another issue. Let's get this working without a manual delay.

Reply:
> If you have to use Task.Delay, you're most likely patching over a deeper issue. Let's get this working without a manual delay.

The delay has been removed, it wasn't masking any deeper issue, it was just there for sanity.


https://user-images.githubusercontent.com/15306909/191082192-a720e7c2-8397-447b-81a6-f87066e60580.mp4